x-pack/plugins/security_solution/public/attack_discovery/pages/index.tsx

```diff
@@ -141,6 +146,13 @@ const AttackDiscoveryPageComponent: React.FC = () => {
     setSelectedConnectorLastUpdated(lastUpdated);
   }, [attackDiscoveries, lastUpdated, replacements]);
 
+  useEffect(() => {
+    // If there is only one connector, set it as the selected connector
+    if (aiConnectors != null && aiConnectors.length === 1) {
+      setConnectorId(aiConnectors[0].id);
+    }
+  }, [aiConnectors, setConnectorId]);
+
```

Comment:
I've somehow found myself in a state where I'm seeing the 'Add Connector' action instead of the 'Connector Selector' when I do have (pre-configured) connectors available. Let me see if I can find steps to reproduce...

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/d72a7e26-9dd5-458b-b7bc-df6bcda138e5" />
</p>

Reply:
I'm seeing similar inconsistent states with the inline connector selector in the Assistant too. I'm wondering if it's related to the changes in this PR.

Reply:
Alrighty, @andrew-goldstein and I paired on this one and looks like this is a pre-existing issue with the ole connector selector flow, however we can get around it by passing in `isFlyoutMode={false}` to `<ConnectorSelectorInline` and hiding the component entirely when `aiConnectors?.length > 0` which is available one component higher in `Header`.

Reply:
updated in [this commit]( https://github.com/elastic/kibana/pull/182904/commits/7f1d99cd1d9047d45db9d1132c689ee12df1a589)